src/field_theory/splitting_field.lean

```diff
@@ -609,11 +609,17 @@ section splitting_field
 def factor (f : K[X]) : K[X] :=
 if H : ∃ g, irreducible g ∧ g ∣ f then classical.some H else X
 
-instance irreducible_factor (f : K[X]) : irreducible (factor f) :=
+lemma irreducible_factor (f : K[X]) : irreducible (factor f) :=
 begin
   rw factor, split_ifs with H, { exact (classical.some_spec H).1 }, { exact irreducible_X }
 end
 
+/-- See note [fact non-instances]. -/
+lemma fact_irreducible_factor (f : K[X]) : fact (irreducible (factor f)) :=
+⟨irreducible_factor f⟩
+
+local attribute [instance] fact_irreducible_factor
+
```

Comment:
Does anything actually use this?

Reply:
Yeah, the whole of the below file needs it for the field instance on `adjoin_root`